src/leiningen/new/chestnut/src/clj/chestnut/server.clj

```diff
@@ -15,12 +21,26 @@
      :body (io/input-stream (io/resource "public/index.html"))})
   (resources "/"))
 
-(def http-handler
-  (-> routes
-      (wrap-defaults {{ring-defaults}})
-      wrap-with-logger
-      wrap-gzip))
+(defn get-http-handler [_]
+  (-> routes))
+
+(defn prod-system []
+  (component/system-map
+   :routes (new-endpoint get-http-handler)
+   :middleware (new-middleware  {:middleware [[wrap-defaults :defaults]
+                                              wrap-with-logger
+                                              wrap-gzip]
+                                 :defaults {{ring-defaults}}})
+   :handler (component/using
+             (new-handler)
+             [:routes :middleware])
+   :http (component/using
+          (new-web-server (Integer. (or (env :port) 10555)))
+          [:handler])))
+
+(defn run-prod []
+  (reloaded.repl/set-init! prod-system)
+  (reloaded.repl/go))
```

Comment:
seems fishy to be using `reloaded-repl` here. Ideally it should be a dev-time dependency only.